pkg/console/subresource/deployment/deployment_test.go

```diff
@@ -366,13 +391,494 @@ func TestDefaultDeployment(t *testing.T) {
 	}
 }
 
-func TestDefaultDownloadsDeployment(t *testing.T) {
+func TestWithAnnotations(t *testing.T) {
+	type args struct {
+		deployment                  *appsv1.Deployment
+		cm                          *corev1.ConfigMap
+		serviceCAConfigMap          *corev1.ConfigMap
+		defaultIngressCertConfigMap *corev1.ConfigMap
+		trustedCAConfigMap          *corev1.ConfigMap
+		sec                         *corev1.Secret
+		proxyConfig                 *configv1.Proxy
+		infrastructureConfig        *configv1.Infrastructure
+	}
+
+	consoleConfig := &corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            "console-config",
+			ResourceVersion: "10245",
+		},
+		Data:       map[string]string{"console-config.yaml": ""},
+		BinaryData: nil,
+	}
+
+	infrastructureConfig := &configv1.Infrastructure{
+		ObjectMeta: metav1.ObjectMeta{
+			ResourceVersion: "12345",
+		},
+		Status: configv1.InfrastructureStatus{
+			InfrastructureTopology: configv1.SingleReplicaTopologyMode,
+		},
+	}
+
+	proxyConfig := &configv1.Proxy{
+		ObjectMeta: metav1.ObjectMeta{
+			ResourceVersion: "54321",
+		},
+	}
+
+	serviceCAConfigMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			ResourceVersion: "34343",
+		},
+	}
+	defaultIngressCertConfigMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			ResourceVersion: "77777",
+		},
+	}
+	trustedCAConfigMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			ResourceVersion: "75577",
+		},
+	}
+
+	sec := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			ResourceVersion: "101010",
+		},
+	}
+
+	tests := []struct {
+		name string
+		args args
+		want *appsv1.Deployment
+	}{
+		{
+			name: "Test Default Annotations",
+			args: args{
+				deployment: &appsv1.Deployment{
+					ObjectMeta: metav1.ObjectMeta{
+						Annotations: map[string]string{},
```

Comment:
@jhadvig one question I had: should these helper functions handle the case where `Deployment.ObjectMeta.Annotations` has not been initialized to an empty string map? Or are we guaranteed these will always be initialized?

Reply:
Since console's deployments are relying on other resources(infraConfig, ingress, various CMs, ....) and generating its deployments from them, we expect them to exist and be available. Otherwise operator gets degraded.